oak_functions/abi/src/lib.rs

```diff
@@ -83,4 +83,37 @@ extern "C" {
         inference_ptr_ptr: *mut *mut u8,
         inference_len_ptr: *mut usize,
     ) -> u32;
+
+    /// Reads a message, i.e., `dest_buf_len_ptr` bytes, from the channel with `channel_handle` into
+    /// the buffer at `dest_buf_ptr_ptr`. After a successful call to `channel_read`
+    /// the buffer at `dest_buf_ptr_ptr` holds bytes encoding a message taken from the specified
+    /// channel.
+    ///
+    /// Returns a status code to indicate success. In particular, returns immediately
+    /// with an appropriate status code, if no message is available on the channel.
+    pub fn channel_read(
+        channel_handle: i32,
```

Comment:
I don't remember what was the conclusion of the discussion between i32 and u32 for handles? I would expect that in the runtime we would represent them as u32 (the sign of the handle does not seem to have a meaning, so i32 seems misleading). Granted, both of them would be represented as i32 in Wasm, but as long as we can use a more precise type here (u32) and Rust converts it to the appropriate underlying type (i32), then I think we should do that. Another reason for using u32 on the runtime side is that one possible way of allocating handles to newly created channels would be to just start from 0 and increase, similar to file descriptors; that also suggests that they should be unsigned.

Reply:
done, thank you!